portfolio/src/main/java/com/google/sps/data/LongestTaskFirstScheduler.java

```diff
@@ -0,0 +1,104 @@
+package com.google.sps.data;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Optional;
+
+public class LongestTaskFirstScheduler implements TaskScheduler {
+
+  // Comparator for sorting tasks by duration in descending order and then by task name
+  // alphabetically.
+  public static final Comparator<Task> sortByTaskDurationDescendingThenName =
+      Comparator.comparing(Task::getDuration).reversed().thenComparing(Task::getName);
+
+  /**
+   * Schedules the tasks so that the longest tasks are scheduled to the first possible free time
+   * range of the day. This approach tries to prioritize long tasks so that they are scheduled.
+   */
+  public Collection<ScheduledTask> schedule(
```

Comment:
Consider clarifying in the method comment that tasks can be modified by this method. Or you can update the method so that you don't modify tasks param, and just copy it into a new list. I think the latter might be a good idea.

Reply:
I think the initialization of `tasksList` is equivalent to copying the tasks into a new list?